pexec/managed_process.go

```diff
@@ -187,7 +197,24 @@ func (p *managedProcess) manage(stdOut, stdErr io.ReadCloser) {
 					}
 					return
 				}
-				p.logger.Debugw("output", "name", name, "data", string(line))
+				if p.shouldLog {
+					p.logger.Debugw("output", "name", name, "data", string(line))
+				}
+				if p.logWriter != nil && !logWriterError {
+					_, err := (*p.logWriter).Write(line)
+					if err == nil {
+						_, err = (*p.logWriter).Write([]byte("\n"))
+					}
+					if err != nil {
+						if !errors.Is(err, io.ErrClosedPipe) {
+							p.logger.Errorw("error writing process output to log writer", "name", name, "error", err)
+						}
+						if !p.shouldLog {
+							return
```

Comment:
this returning doesn't seem right and like a bug

Reply:
that's fine. it's better than stopping the process. I think that needs to be an option if an error logging should quit. in practice failing to write to an io.Writer should never happen

Reply:
Returning here won't stop the managed process, just the goroutine responsible for logging. And I do expect to fail to write to the io.Writer for slam, since slam closes the reader/writer pair, once it retrieves the port.

Reply:
oh duh! sure, great idea then